astropy/io/fits/hdu/hdulist.py

```diff
@@ -114,6 +114,12 @@ def fitsopen(name, mode='readonly', memmap=None, save_backup=False,
             If `True`, image data is not scaled using BSCALE/BZERO values
             when read.
 
+        - **return_bytes** : bool
```

Comment:
I wonder about the name `return_bytes`.  It might not be immediately obvious to code readers what that implies.  So maybe `string_as_bytes` or `character_as_bytes` (bool) or `character_type = ('bytes'|'unicode')`.

Reply:
Can we go a bit beyond that and follow numpy in adding an `encoding` argument (https://github.com/numpy/numpy/pull/4208)? This especially since table columns can now handle at least encoding `utf-8` (`U` would correspond to `utf-32`, which would be the default; `bytes` and `utf-8` would do the same thing - ideally we change `Column` to have encoding as an option...)

Reply:
Maybe `try_decode_bytes` or `try_decode_ascii` ? (Only ascii columns are decoded)
Not sure about adding an encoding argument, it can be misleading.

Reply:
Can FITS columns actually include non-ASCII data though?

Reply:
I went with ``character_as_bytes`` since that is the most intuitive in my opinion